### PR TITLE
feat: support warning_markdown on metrics

### PR DESCRIPTION
**Pull request by etr2460**
🏆 Enhancements
Precursor to https://github.com/apache/superset/pull/13606, supports both warning_text and warning_markdown side by side in the metric option component.

Tested by npm linking to superset:
Screen Shot 2021-03-17 at 10 33 39 AM
Screen Shot 2021-03-17 at 10 33 31 AM
Screen Shot 2021-03-17 at 10 33 23 AM

### CODE REVIEW

File: packages/superset-ui-core/src/components/SafeMarkdown.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
```
Comment (etr2460) [inline packages/superset-ui-core/src/components/SafeMarkdown.tsx:1]: Moved from apache/superset, will delete from that codebase when I bump the superset-ui-core version used in superset

File: packages/superset-ui-core/package.json [hunk 2]
```diff
@@ -55,6 +55,7 @@
     "lodash": "^4.17.11",
     "pretty-ms": "^7.0.0",
     "react-error-boundary": "^1.2.5",
+    "react-markdown": "^4.3.1",
```
Comment (ktmud) [inline packages/superset-ui-core/package.json:58]: Could you update `yarn.lock` as well?

File: packages/superset-ui-core/src/components/SafeMarkdown.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import ReactMarkdown, { MarkdownAbstractSyntaxTree } from 'react-markdown';
+// @ts-ignore no types available
+import htmlParser from 'react-markdown/plugins/html-parser';
+
+import { FeatureFlag, isFeatureEnabled } from '../index';
```
Comment (ktmud) [inline packages/superset-ui-core/src/components/SafeMarkdown.tsx:6]:
```suggestion
import { FeatureFlag, isFeatureEnabled } from '../utils';
```

In general I think imports within a library should be more explicit to save any future trouble of circular imports.

**Review by ktmud — APPROVED** [review]
LGTM.

File: packages/superset-ui-core/package.json (hunk 2 above)
Comment (etr2460) [reply packages/superset-ui-core/package.json:58]: yarn.lock is git ignored for me. or do you mean the one in the root directory?

Comment (etr2460) [reply packages/superset-ui-core/package.json:58]: ah, i see the yarn.lock in root is updated a lot. i'll do that

**Pull request merged**